qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -310,35 +310,34 @@ def amplify(self, amplification_problem: AmplificationProblem) -> "GroverResult"
                 }
                 top_measurement, max_probability = max(circuit_results.items(), key=lambda x: x[1])
 
+            elif self._quantum_instance.is_statevector:
+                qc = self.construct_circuit(amplification_problem, power, measurement=False)
+                circuit_results = self._quantum_instance.execute(qc).get_statevector()
+                num_bits = len(amplification_problem.objective_qubits)
+
+                # trace out work qubits
+                if qc.width() != num_bits:
+                    indices = [
+                        i
+                        for i in range(qc.num_qubits)
+                        if i not in amplification_problem.objective_qubits
+                    ]
+                    rho = partial_trace(circuit_results, indices)
+                    circuit_results = np.diag(rho.data)
+
+                max_amplitude = max(circuit_results.max(), circuit_results.min(), key=abs)
+                max_amplitude_idx = np.where(circuit_results == max_amplitude)[0][0]
+                top_measurement = np.binary_repr(max_amplitude_idx, num_bits)
+                max_probability = np.abs(max_amplitude) ** 2
+                shots = 1
             else:
```

Comment:
The use of quantum_instance with grover is deprecated - this is if/else under the else branch depending on the the simulation type (statevector or not). This code will be removed in the note too distant future - the whole of that else branch that you have moved out into an elif and else. My preference would be to keep it as it was since with the outer if/else it, to me, separated more cleanly into two parts the new code that will be kept and the deprecated code that will end up being removed.